src/EditorFeatures/CSharpTest/RemoveUnnecessaryParentheses/RemoveUnnecessaryParenthesesTests.cs

```diff
@@ -1053,5 +1053,470 @@ void M(string s)
+        public async Task TestNullableCastNoAmbiguity1()
+        {
+            await TestAsync(
+@"class C
+{
+    void M()
+    {
+        int x = (T?)$$(-1);
+    }
+}",
+@"class C
+{
+    void M()
+    {
+        int x = (T?)-1;
+    }
+}", offeredWhenRequireForClarityIsEnabled: true);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsRemoveUnnecessaryParentheses)]
+        public async Task TestNullableCastNoAmbiguity2()
+        {
+            await TestAsync(
+@"class C
+{
+    void M()
+    {
+        int x = (T?)$$(+1);
+    }
+}",
+@"class C
+{
+    void M()
+    {
+        int x = (T?)+1;
+    }
+}", offeredWhenRequireForClarityIsEnabled: true);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsRemoveUnnecessaryParentheses)]
+        public async Task TestNullableCastNoAmbiguity3()
+        {
+            await TestAsync(
+@"class C
+{
+    void M()
+    {
+        int x = (T?)$$(&x);
+    }
+}",
+@"class C
+{
+    void M()
+    {
+        int x = (T?)&x;
+    }
+}", offeredWhenRequireForClarityIsEnabled: true);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsRemoveUnnecessaryParentheses)]
+        public async Task TestNullableCastNoAmbiguity4()
+        {
+            await TestAsync(
+@"class C
+{
+    void M()
+    {
+        int x = (T?)$$(*x);
+    }
+}",
+@"class C
+{
+    void M()
+    {
+        int x = (T?)*x;
+    }
+}", offeredWhenRequireForClarityIsEnabled: true);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsRemoveUnnecessaryParentheses)]
+        public async Task TestAliasCastNoAmbiguity1()
+        {
+            await TestAsync(
+@"class C
+{
+    void M()
+    {
+        int x = (e::N.T)$$(-1);
+    }
+}",
+@"class C
+{
+    void M()
+    {
+        int x = (e::N.T)-1;
+    }
+}", offeredWhenRequireForClarityIsEnabled: true);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsRemoveUnnecessaryParentheses)]
+        public async Task TestAliasCastNoAmbiguity2()
+        {
+            await TestAsync(
+@"class C
+{
+    void M()
+    {
+        int x = (e::N.T)$$(+1);
+    }
+}",
+@"class C
+{
+    void M()
+    {
+        int x = (e::N.T)+1;
+    }
+}", offeredWhenRequireForClarityIsEnabled: true);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsRemoveUnnecessaryParentheses)]
+        public async Task TestAliasCastNoAmbiguity3()
+        {
+            await TestAsync(
+@"class C
+{
+    void M()
+    {
+        int x = (e::N.T)$$(&x);
+    }
+}",
+@"class C
+{
+    void M()
+    {
+        int x = (e::N.T)&x;
+    }
+}", offeredWhenRequireForClarityIsEnabled: true);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsRemoveUnnecessaryParentheses)]
+        public async Task TestAliasCastNoAmbiguity4()
+        {
+            await TestAsync(
+@"class C
+{
+    void M()
+    {
+        int x = (e::N.T)$$(*x);
```

Comment:
nit: these are all in a different order (+-*&)

Reply:
Each goes in the same order afaict.

Reply:
correction: only TestCastAmbiguity is in different order (+-&*), the rest is -+&*

Reply:
#resolved